### ISSUE TITLE
List Of PayPal Changes

### ISSUE THREAD

**Issue by sierrab123** [ingenesis/shopp#3411]
Just raising this as an issue to ensure we are on top of this

What changes are being made and when?

The following are technical changes that may require some upgrades to your system. Please share this information with your development team or hosting provider.

    Now

        Test Sandbox endpoints and tlstest.paypal.com are live.
        New SFTP IP addresses are live.
        Sandbox is issuing API Credential Certificates with new standard (2048-bit, SHA-256).
    Jan 14, 2016
    After this date, Sandbox API endpoints only support new standard (HTTP/1.1, TLS 1.2 and SHA-256 certificates).

        This includes http://www.sandbox.paypal.com only accepting HTTPS for IPN Postbacks.
    Jan 31, 2016

        Production starts issuing API Credential Certificates with new standard (2048-bit, SHA-256).
    Feb 29, 2016

        Test Sandbox endpoints will be removed.
    Mar 17, 2016

        New SFTP IP addresses add to DNS for reports.paypal.com.
    Apr 14, 2016

        Old SFTP IP addresses removed from DNS for reports.paypal.com.
    May 12, 2016

        Old SFTP IP addresses stop working.
    Jun 17, 2016

        After this date, Production API endpoints will start moving to the new standard (HTTP/1.1, TLS 1.2 and SHA-256 certificates)
    Sep 30, 2016

        IPN postbacks to http://www.paypal.com only allow HTTPS
    Jan 1, 2018

        All Certificate API Credentials must have been upgraded to the new standard.

To help you navigate through these technical changes, we created the [2016 Merchant Security Roadmap](https://www.paypal-knowledge.com/infocenter/> index?page=content&id=FAQ1913). The website offers detailed information about each of the upcoming changes, including dates when these changes are scheduled* and > security best practices.
What do I need to do now?

While there’s still some time before these changes go into effect, here’s what you can do now. If you’re not using a hosted shopping cart or partner, please consider > doing the following to prepare for these changes:

Incorporate this work into your 2016 technology update plans by engaging with your technical or web development team.
To see if you’re already compatible with these security upgrades, test your configuration in the PayPal Sandbox.
How do I know if I’m already compatible with these security upgrades?

You can test your system now! We’ve created new, temporary Sandbox endpoints that are configured with the latest security standards. Go to the PayPal 2016 Merchant > Security Roadmap for easy-to-follow instructions on how to test endpoints in the Sandbox > environment today.

If you’re not sure what upgrades your system may require, no worries – we’ve got that covered. We’ll write you again in January with more specific details about what > these changes mean to your system. In the meantime, we encourage you to review the [PayPal 2016 Merchant Security Roadmap](https://www.paypal-knowledge.com/infocenter> /index?page=content&id=FAQ1913). For questions, please go to the Contact page on PayPal.com.

Thank you for your support of our commitment to maintaining high security standards for all of our global customers.

Scheduled change dates provided in this email and the PayPal 2016 Merchant Security Roadmap are subject to change. You’ll be notified immediately of any changes to these plans.

**Comment by jonathandavis** [comment]
TL;DR; The net effect on Shopp is nothing.

> Test Sandbox endpoints and tlstest.paypal.com are live.
>    New SFTP IP addresses are live.

Does not affect Shopp.

>    Sandbox is issuing API Credential Certificates with new standard (2048-bit, SHA-256).
> Jan 14, 2016
> After this date, Sandbox API endpoints only support new standard (HTTP/1.1, TLS 1.2 and SHA-256 certificates).
>    This includes http://www.sandbox.paypal.com only accepting HTTPS for IPN Postbacks.
Jan 31, 2016

Does not affect Shopp. Server/browser support required. Shopp already uses HTTPS for IPN Postbacks.

>    Production starts issuing API Credential Certificates with new standard (2048-bit, SHA-256).

Does not affect Shopp. Again, server/browser support for SHA-2 certs is required.

> Feb 29, 2016
>    Test Sandbox endpoints will be removed.

Has no effect on live transactions, only affects testing transactions with the sandbox.

> Mar 17, 2016
>    New SFTP IP addresses add to DNS for reports.paypal.com.
> Apr 14, 2016
>    Old SFTP IP addresses removed from DNS for reports.paypal.com.
> May 12, 2016
>    Old SFTP IP addresses stop working.

None of this affects Shopp.

> Jun 17, 2016
>    After this date, Production API endpoints will start moving to the new standard (HTTP/1.1, TLS 1.2 and SHA-256 certificates)
> Sep 30, 2016
>    IPN postbacks to http://www.paypal.com only allow HTTPS
> Jan 1, 2018
>    All Certificate API Credentials must have been upgraded to the new standard.

Again, this is related to server/browser support for SHA-2 certificates. Does not affect Shopp itself.

In summary, none of this requires any changes in Shopp to accommodate them. It is up to hosting providers and browser vendors to provide support.
